src/core/app/actions/streaming.rs

```diff
@@ -175,6 +189,42 @@ fn prepare_retry_stream(app: &mut App, ctx: AppActionContext) -> Option<AppComma
     }
 }
 
+fn prepare_refine_stream(
```

Comment:
We actually need to include the message the assistant originally generated in the transcript submitted to the API on the REFINE request. It's just that it's immediately omitted from the transcript and not sent subsequently. Otherwise the API cannot actually see the message it's supposed to refine.